fix(ZNTA-2313): update admin rejection panel

kathryngo opened the pull request:
JIRA issue URL: https://zanata.atlassian.net/browse/ZNTA-2313

Removed the accordion with the text input rows. This is kind of confusing since these can't be used as a starting point without copying the text so the examples have been included as text under the description.

The toggle editable button is removed as it isn't necessary. If admins want to change the criteria they can just edit it, and if they aren't happy with their changes they shouldn't save them.

Also fixed padding and spacing in the rows and added breadcrumbs as requested.

## QA 
Please test the functionality of the admin Reject Criteria page in frontend

## Initial page
<img width="1369" alt="1" src="https://user-images.githubusercontent.com/18179401/35136565-0c4139d0-fd30-11e7-9090-9b32137a5e6b.png">

## Add first criteria
<img width="1382" alt="2" src="https://user-images.githubusercontent.com/18179401/35136574-199e1a58-fd30-11e7-9098-501682a628dd.png">

## Add second criteria
<img width="1380" alt="3" src="https://user-images.githubusercontent.com/18179401/35136583-22e41bbc-fd30-11e7-9706-48c7a568dbd1.png">

## Modal - criteria added
<img width="1378" alt="reject-withcriteria" src="https://user-images.githubusercontent.com/18179401/35315159-9afbfcc0-0116-11e8-9875-d007d58709ad.png">

## Modal - no criteria added
<img width="1381" alt="nocriteria" src="https://user-images.githubusercontent.com/18179401/35315188-acbacef0-0116-11e8-93d8-05071fcddd01.png">



## Checklist

- JIRA link
- Check target branch
- Make sure all commit statuses are green or otherwise documented reasons to ignore
- QA needs to evaluate against the JIRA ticket
- Changed files and commits make sense for this PR

See [Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines) more for information.

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/zanata/zanata-platform/672)
<!-- Reviewable:end -->

efloden commented:
Reviewed 1 of 3 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/672)*
<!-- Sent from Reviewable.io -->

efloden commented:
:lgtm:

djansen-redhat commented:
- [ ] The "editable toggle" I *believe* actually referred to the use of the option in the editor - in that it means text in the rejection field will be appended to the criteria reason. If not editable, then that text will be ignored.
- [ ] No indication of save success
- [ ] 500 Server error on attempting to delete a used criteria (!?)
- [ ] No field max sizing, 500 server error on too much text
- [ ] No title/tooltip for 'Delete criteria'
- [ ] TextArea can be resized beyond the parent container, resulting in clipped buttons. Should just be a TextField anyhow.
- [ ] Can save a criteria without text
- [ ] The parentheses seem unnecessary, it could just be eg **Translation Errors:** terminology, mistranslated, etc
- [ ] Not sure if it matters at all, but the design is dissimilar to the "centred column" of the other admin pages
- [ ] Deletion confirmation?

Not directly related:
- [ ] The amount of text that can be entered causes the dropdown in the editor to extend well past the page
- [ ] Empty text still shows in the dropdown

kathryngo commented:
The "editable toggle" I believe actually referred to the use of the option in the editor - in that it means text in the rejection field will be appended to the criteria reason. If not editable, then that text will be ignored. 
- is this even necessary? Seems to overcomplicate the UI and surely the field can be ignored if there is not text and any text in the rejection field will be appended anyways. 

 No indication of save success
- use Alert - success to do this 
- *Kathryn to add example to storybook*

 500 Server error on attempting to delete a used criteria (!?)
- good luck with that one

 No field max sizing, 500 server error on too much text
- *Kathryn to fix*

 No title/tooltip for 'Delete criteria'
- *Kathryn to fix*

 TextArea can be resized beyond the parent container, resulting in clipped buttons. Should just be a TextField anyhow.
- *Kathryn to fix*

 Can save a criteria without text
- another one for the engineers

 The parentheses seem unnecessary, it could just be eg Translation Errors: terminology, mistranslated, etc
- *Kathryn to fix*

 Not sure if it matters at all, but the design is dissimilar to the "centred column" of the other admin pages
- more content, needs a wider view

 Deletion confirmation?
-  *Kathryn to fix*

Not directly related:

 The amount of text that can be entered causes the dropdown in the editor to extend well past the page
- *Kathryn to fix*

 Empty text still shows in the dropdown
- Unsure what I can do about this one

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/672#-:-L3WQeSfcOQ_W9gany_g:b-9nt7s7)*
<!-- Sent from Reviewable.io -->

djansen-redhat commented:
>> The "editable toggle" I believe actually referred to the use of the option in the editor - in that it means text in the rejection field will be appended to the criteria reason. If not editable, then that text will be ignored.

> is this even necessary? Seems to overcomplicate the UI and surely the field can be ignored if there is not text and any text in the rejection field will be appended anyways.

To be honest, I'm not sure. My thought would have been to make it less database-y, in that it simply copies the 'criteria' to the reject reason. If the reviewer wants to change it, then it's there to change.

kathryngo commented:
@djansen-redhat what copying? in the interface the example criteria are presented as text and the text inputs are empty by default. So there needs to be editing done to these fields automatically. Is there something I am missing?

kathryngo commented:
@djansen-redhat except for that first issue and those best left to engineers I have completed the changes based on your feedback. I have done the following:

No indication of save success
I added an example to storybook under the Rejection Form

No field max sizing, 500 server error on too much text
Fixed

No title/tooltip for 'Delete criteria'
Fixed

TextArea can be resized beyond the parent container, resulting in clipped buttons. Should just be a TextField anyhow.
Fixed

The parentheses seem unnecessary, it could just be eg Translation Errors: terminology, mistranslated, etc
Fixed

Deletion confirmation?
Added a story to Rejections Form

The amount of text that can be entered causes the dropdown in the editor to extend well past the page
Fixed

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/672#-:-L3_Q0C_0fOnekFYMrYS:bon2mvi)*
<!-- Sent from Reviewable.io -->

kathryngo commented:
Ok, so @djansen-redhat and I sat down this afternoon and came to the following conclusions:

**When there is >= 1 criteria**, an 'Other' option should be added to the dropdown. 
Comments are not required however.

**When no criteria has been set**, no dropdown appears as there is no criteria option to select. Comments are now required so the reviewer can state why they have rejected the translation.

Please see the images attached to the description in this PR, and also the RejectTranslationModal stories in editor. 

Please note, I had to create a new js file (RejectTranslationsModalNoCrit.js) so that I could add the other modal example to storybook.

efloden commented:
Reviewed 2 of 3 files at r3, 1 of 1 files at r4, 5 of 5 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[server/zanata-frontend/src/app/editor/components/RejectTranslationModal/RejectTranslationModalNoCrit.js, line 15 at r5](https://reviewable.io:443/reviews/zanata/zanata-platform/672#-L3aron8v0Ic3t8nozjM:-L3aron8v0Ic3t8nozjN:bx8ow4e) ([raw file](https://github.com/zanata/zanata-platform/blob/f535e228ea0786aca40baa2cc1431b5bc57f697c/server/zanata-frontend/src/app/editor/components/RejectTranslationModal/RejectTranslationModalNoCrit.js#L15)):*
> ```JavaScript
> export const CRITICAL = 'Critical'
> ```

We probably wont want an extra component for the modal without the criteria in practice, as it will mean a lot of duplicated code

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/672)*
<!-- Sent from Reviewable.io -->

kathryngo commented:
Review status: 7 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[server/zanata-frontend/src/app/editor/components/RejectTranslationModal/RejectTranslationModalNoCrit.js, line 15 at r5](https://reviewable.io:443/reviews/zanata/zanata-platform/672#-L3aron8v0Ic3t8nozjM:-L3bcNH7PVuuvqFPlAvp:bu2dm8x) ([raw file](https://github.com/zanata/zanata-platform/blob/f535e228ea0786aca40baa2cc1431b5bc57f697c/server/zanata-frontend/src/app/editor/components/RejectTranslationModal/RejectTranslationModalNoCrit.js#L15)):*
<details><summary><i>Previously, efloden (Earl Floden) wrote…</i></summary><blockquote>

We probably wont want an extra component for the modal without the criteria in practice, as it will mean a lot of duplicated code
</blockquote></details>

Done.  Also removed unused constant from storybook file above.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/672)*
<!-- Sent from Reviewable.io -->

efloden commented:
Reviewed 2 of 2 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/672)*
<!-- Sent from Reviewable.io -->

djansen-redhat commented:
@huangp @seanf I think you should both be part of this discussion, as it may conflict with specification decided upon by yourselves and the translators. Specifically there is no concept of 'editable' - all criteria can be elaborated upon.

Furthermore, I am concerned that this feature is not well presented. It is somewhat invisible until the reviewer has already made a decision to reject the translation.
Review criteria should be something the reviewer can see up front, to make a judgement on text flows based on said criteria - not unlike our Github reviewer checklist.
These criteria may also be in conflict with the project maintainer's workflow or expectations. I believe it could be better provided as a project feature.

kathryngo commented:
@efloden I just added the toggle back to the admin page so that comments can be set as required. 

![rejsettings](https://user-images.githubusercontent.com/18179401/35956309-89e9f290-0ce0-11e8-9c38-2870fff8b417.png)

efloden commented:
Reviewed 1 of 2 files at r7, 6 of 6 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/672)*
<!-- Sent from Reviewable.io -->

efloden commented:
Review status: 4 of 9 files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

*[server/zanata-frontend/src/app/editor/components/RejectTranslationModal/RejectTranslationModalNoCrit.js, line 6 at r9](https://reviewable.io:443/reviews/zanata/zanata-platform/672#-L5klhYvcdXZOkyHJoOt:-L5klhYvcdXZOkyHJoOu:b-u8gykl) ([raw file](https://github.com/zanata/zanata-platform/blob/5345e23d2f49e96bedc204b0e8a475aafb222daf/server/zanata-frontend/src/app/editor/components/RejectTranslationModal/RejectTranslationModalNoCrit.js#L6)):*
> ```JavaScript
> import Button from '../Button'
> import Dropdown from '../Dropdown'
> import { Row } from 'react-bootstrap'
> ```

Some unused imports and variables appear to cause the build failure in this file: (Dropdown, Icon, key, className, onHide, isOpen etc in the build logs)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/672)*
<!-- Sent from Reviewable.io -->

kathryngo commented:
Review status: 8 of 10 files reviewed at latest revision, 1 unresolved discussion.

---

*[server/zanata-frontend/src/app/editor/components/RejectTranslationModal/RejectTranslationModalNoCrit.js, line 6 at r9](https://reviewable.io:443/reviews/zanata/zanata-platform/672#-L5klhYvcdXZOkyHJoOt:-L5pa6JHFeN1ze1a3GB5:b-896fix) ([raw file](https://github.com/zanata/zanata-platform/blob/5345e23d2f49e96bedc204b0e8a475aafb222daf/server/zanata-frontend/src/app/editor/components/RejectTranslationModal/RejectTranslationModalNoCrit.js#L6)):*
<details><summary><i>Previously, efloden (Earl Floden) wrote…</i></summary><blockquote>

Some unused imports and variables appear to cause the build failure in this file: (Dropdown, Icon, key, className, onHide, isOpen etc in the build logs)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/672)*
<!-- Sent from Reviewable.io -->

efloden commented:
Reviewed 1 of 1 files at r10, 1 of 1 files at r11.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/672)*
<!-- Sent from Reviewable.io -->

djansen-redhat commented on server/zanata-frontend/src/app/components/RejectionsForm/index.js:
```diff
@@ -113,13 +116,15 @@ class RejectionsForm extends Component {
-    const commentToggle = isAdminMode ? (
-      <FormGroup controlId='formInlineEditable'>
-        <ControlLabel>Editable</ControlLabel><br />
+    const editableToggle = isAdminMode ? (
```
Defined but not used

kathryngo commented:
Review status: 5 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[server/zanata-frontend/src/app/components/RejectionsForm/index.js, line 125 at r12](https://reviewable.io:443/reviews/zanata/zanata-platform/672#-L36mTfH3ef7VXRB2pC3-r12-125:-L5vzg69a1A6T1zUUetm:b-896fix) ([raw file](https://github.com/zanata/zanata-platform/blob/a5e569ccc1b94663a838b0bd852d951f1ec1d710/server/zanata-frontend/src/app/components/RejectionsForm/index.js#L125)):*
<details><summary><i>Previously, djansen-redhat (Damian Jansen) wrote…</i></summary><blockquote>

eslint error: editableToggle is defined but not used
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/672)*
<!-- Sent from Reviewable.io -->

spathare21 commented:
@kathryngo 
- Seems like few issues mentioned by Damian in above comment are not fixed 
https://github.com/zanata/zanata-platform/pull/672#issuecomment-359309253

- 500 internal server error on deleting Criteria 

![screenshot from 2018-02-22 12-13-05](https://user-images.githubusercontent.com/23230884/36523977-3884d2fe-17ca-11e8-8319-29b50edcf6c0.png)

- 400 bad request on adding review criteria which contains lot of text (there should be limitation of characters in review criteria text box)


![screenshot from 2018-02-22 12-13-17](https://user-images.githubusercontent.com/23230884/36523982-3cfaf32c-17ca-11e8-9171-7d970c52f4d3.png)

djansen-redhat commented:
Reviewed 2 of 4 files at r12, 1 of 2 files at r13, 2 of 2 files at r14.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/672)*
<!-- Sent from Reviewable.io -->

djansen-redhat commented:
@spathare21 at this present moment the Operation Failed notice can stay. We can consider a fix later.
@kathryngo @efloden agree?

kathryngo commented:
@djansen-redhat agreed

kathryngo commented:
@djansen-redhat ready to verify?

spathare21 commented:
@djansen-redhat  ok thanks then verifying it.

kathryngo commented:
will merge after full func. tests are run

djansen-redhat commented:
Reviewed 1 of 2 files at r15.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/672)*
<!-- Sent from Reviewable.io -->
